### PR TITLE
Feature/home controller

### PR DESCRIPTION
**Pull request by KarolisM**
Kontroleris homepage visiem trim vartotojam

### CODE REVIEW

File: src/MasterPeace/Bundle/UpReadBundle/Controller/HomePageController.php [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Route;
+use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+
+class HomePageController extends Controller
+{
+    /**
+     * @Route("/", name="homepage")
+     *
+     * @return \Symfony\Component\HttpFoundation\Response
```
Comment (povils) [inline src/MasterPeace/Bundle/UpReadBundle/Controller/HomePageController.php]: @return Response

File: src/MasterPeace/Bundle/UpReadBundle/Controller/HomePageController.php [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+     */
+    public function redirectUserAction()
+    {
+        $response = null;
```
Comment (povils) [inline src/MasterPeace/Bundle/UpReadBundle/Controller/HomePageController.php]: trink lauk

File: src/MasterPeace/Bundle/UpReadBundle/Controller/HomePageController.php [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+        if ($this->getUser()->isTeacher()) {
```
Comment (povils) [inline src/MasterPeace/Bundle/UpReadBundle/Controller/HomePageController.php]: tiesiog jei studentas returnink i student.html.twig antraip i teacher - nereik to admin jum

File: src/MasterPeace/Bundle/UpReadBundle/Controller/HomePageController.php [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+        return $response;
+    }
+
+    /**
+     * @Route("/admin", name="admin")
+     *
+     * @return \Symfony\Component\HttpFoundation\Response
+     */
+    public function redirectAdminAction()
```
Comment (povils) [inline src/MasterPeace/Bundle/UpReadBundle/Controller/HomePageController.php]: nesuprantu kam cia tas? jei admin bundle turit

File: src/MasterPeace/Bundle/UpReadBundle/Resources/views/admin.html.twig [hunk 5]
```diff
@@ -0,0 +1 @@
+admin
```
Comment (povils) [inline src/MasterPeace/Bundle/UpReadBundle/Resources/views/admin.html.twig]: trinam lauk si faile

File: src/MasterPeace/Bundle/UpReadBundle/Tests/Controller/HomePageControllerControllerTest.php [hunk 6]
```diff
@@ -0,0 +1,9 @@
+<?php
+
+namespace MasterPeace\Bundle\UpReadBundle\Tests\Controller;
+
+use Symfony\Bundle\FrameworkBundle\Test\WebTestCase;
+
+class HomePageControllerControllerTest extends WebTestCase
+{
```
Comment (povils) [inline src/MasterPeace/Bundle/UpReadBundle/Tests/Controller/HomePageControllerControllerTest.php]: tai ko testa nepabaigei?:)

File: app/config/security.yml [hunk 7]
```diff
@@ -44,6 +44,5 @@ security:
         - { path: ^/login$, role: IS_AUTHENTICATED_ANONYMOUSLY }
         - { path: ^/register, role: IS_AUTHENTICATED_ANONYMOUSLY }
         - { path: ^/resetting, role: IS_AUTHENTICATED_ANONYMOUSLY }
-        # TODO add separate paths for student and teacher users
-        - { path: ^/*, role: [ROLE_ADMIN] }
-        - { path: ^/admin/, role: ROLE_ADMIN }
+        - { path: ^/$, role: [ROLE_STUDENT, ROLE_TEACHER, ROLE_ADMIN] }
```
Comment (KarolisM) [inline app/config/security.yml]: Teisingai, turi pakakti

File: src/MasterPeace/Bundle/UpReadBundle/Controller/HomePageController.php [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+class HomePageController extends Controller
+{
+    /**
+     * @Route("/", name="homepage")
+     *
+     * @return Response
+     */
+    public function redirectAction()
+    {
+        $response = null;
```
Comment (povils) [inline src/MasterPeace/Bundle/UpReadBundle/Controller/HomePageController.php]: tas $response = null ,nereikia gal. Galima perrasyti taip:

        if ($this->getUser()->isStudent()) {
            return $this->render('MasterPeaceUpReadBundle:HomePage:student.html.twig');
        }

        return $this->render('MasterPeaceUpReadBundle:HomePage:teacher.html.twig');

File: tests/MasterPeace/Bundle/UserBundle/Traits/FormLogInSimulation.php [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+        $crawler = $client->request('GET', $route);
+        $crawler = $client->followRedirect();
+
+        $form = $crawler->selectButton('_submit')->form([
+            '_username' => $username,
+            '_password' => $password,
+        ]);
+
+        $client->submit($form);
+        $crawler = $client->followRedirect();
```
Comment (povils) [inline tests/MasterPeace/Bundle/UserBundle/Traits/FormLogInSimulation.php]: iskart returnink

File: tests/MasterPeace/Bundle/UserBundle/Traits/FormLogInSimulation.php [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+{
+    /**
+     * @param Client $client
+     * @param string $username
+     * @param string $password
+     * @param string $route
+     *
+     * @return \Symfony\Component\DomCrawler\Crawler
+     */
+    public function formLogIn(Client $client, $username, $password, $route)
```
Comment (povils) [inline tests/MasterPeace/Bundle/UserBundle/Traits/FormLogInSimulation.php]: string $username 
string $password
strin $route

File: tests/MasterPeace/Bundle/UserBundle/Traits/LogInSimulation.php [hunk 11]
```diff
@@ -12,7 +12,7 @@
      * @param Client $client
      * @param array $roles
      * @param string $username
-     * @param string $password
+     * @param null $password
```
Comment (povils) [inline tests/MasterPeace/Bundle/UserBundle/Traits/LogInSimulation.php]: string|null

File: src/MasterPeace/Bundle/UpReadBundle/Controller/HomePageController.php (hunk 8 above)
Comment (KarolisM) [inline src/MasterPeace/Bundle/UpReadBundle/Controller/HomePageController.php]: Pasirodo galima :+1:

File: tests/MasterPeace/Bundle/UserBundle/Traits/FormLogInSimulation.php [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+
+trait FormLogInSimulation
+{
+    /**
+     * @param Client $client
+     * @param string $username
+     * @param string $password
+     * @param string $route
+     *
+     * @return \Symfony\Component\DomCrawler\Crawler
```
Comment (KarolisM) [inline tests/MasterPeace/Bundle/UserBundle/Traits/FormLogInSimulation.php]: Kodėl editorius man pabrėžia, kad return type tada ne toks kokio tikisi, bet kai nurodomas visas namespace, taip, tada tvarkoje?

File: tests/MasterPeace/Bundle/UserBundle/Traits/LogInSimulation.php (hunk 11 above)
Comment (KarolisM) [inline tests/MasterPeace/Bundle/UserBundle/Traits/LogInSimulation.php]: :+1:

**Pull request merged**